src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSServiceHelper.cs

```diff
@@ -114,4 +114,31 @@ internal static string GetAWSOperationName(IRequestContext requestContext)
         var operationName = Utils.RemoveSuffix(completeRequestName, suffix);
         return operationName;
     }
+
+    internal static string? GetDynamoDbCloudRegion(IRequestContext requestContext)
+    {
+        string? region = requestContext.ClientConfig?.RegionEndpoint?.SystemName;
+
+        // Use ServiceURL as fallback.
+        region ??= GetCloudRegionFromDynamoServiceUrl(requestContext.ClientConfig?.ServiceURL);
+        return region;
+    }
+
+    private static string? GetCloudRegionFromDynamoServiceUrl(string? serviceUrl)
```

Comment:
I think this URL parsing should not be necessary. The region should be available directly in the parameters/request context. Maybe you can dig up the old code that used to collect the `aws.region` before it was removed?

Then we should also consider adding this generically to all calls instead of only for DynamoDB.

Reply:
> I think the fallback to URL still could be required.

Ok makes sense.
Still, it should be possible to restore the removed functionality with changed attribute name as it was before https://github.com/open-telemetry/opentelemetry-dotnet-contrib/commit/131ccf35efcfd21584d7daecf2bbd2c4de196fcc#diff-639d7a5b1e9f3830007b96156d189aaf80d457867122f759ba269adb8387066e (check `src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs` on that commit). This way it would be available for all services not only DynamoDB.

Reply:
I added code for setting of `cloud.region` to the common place, i.e. method `ProcessBeginRequest(...)` where there was removed `aws.region`. It will result in setting of `cloud.region` not only for DynamoDB Client, but also for all other AWS SDK Clients.